### PR TITLE
Changes in README.md

### PR DESCRIPTION
**Pull request by faizanwasif**
### Why are these changes needed?
These changes are necessary to improve the overall clarity, consistency, and professionalism of the README.md documentation for the AutoGen project. The modifications include correcting grammatical errors, fixing typos, and enhancing formatting throughout the document. These improvements will help users better understand the framework’s features, installation instructions, and usage examples, ultimately enhancing their experience with AutoGen.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,11 +29,11 @@ AutoGen streamlines AI development and research, enabling the use of multiple la
 AutoGen offers the following key features:
 
 - **Asynchronous Messaging**: Agents communicate via asynchronous messages, supporting both event-driven and request/response interaction patterns.
-- **Full type support**: use types in all interfaces and enforced type check on build, with a focus on quality and cohesiveness
+- **Full Type Support**: Use types in all interfaces with enforced type checks on build, focusing on quality and cohesiveness.
```
Comment (jackgerrits) [inline README.md:32]:
```suggestion
- **Full Type Support**: Use types in all interfaces, enforced type checks on build, and a focus on quality and cohesiveness.
```

**Review by jackgerrits — CHANGES_REQUESTED** [review]
There are some good edits here - but can you please revert the changes where you are condensing multiple lines into 1. This makes the diff unnecessarily large and hard to see the content changes.

**Comment by faizanwasif** [comment]
@microsoft-github-policy-service agree

**Comment by jackgerrits** [comment]
There are still a large number white space and new line changes, can you please revert those

**Comment by jackgerrits** [comment]
I still see many changes around removing new lines that are not needed for the proposed content changes. For example the entire legal section. If you would like to resolve *all* such changes then we can consider this PR.

File: README.md [hunk 2]
```diff
@@ -285,58 +272,62 @@ AutoGen 0.2 can be installed with:
...
+We are unable to use Discord for project discussions. 
+Therefore, we request that all discussions take place on <https://github.com/microsoft/autogen/discussions/> going forward.
 
 ### What about forks?
 
-<https://github.com/microsoft/autogen/> remains the only official repo for development and support of AutoGen.
-We are aware that there are thousands of forks of AutoGen, including many for personal development and startups building with or on top of the library. We are not involved with any of these forks and are not aware of any plans related to them.
+https://github.com/microsoft/autogen/ remains the only official repo for development and support of AutoGen.
+We are aware that there are thousands of forks of AutoGen, including many for personal development and startups building with or on top of the library. 
+We are not involved with any of these forks and are not aware of any plans related to them.
```
Comment (ekzhu) [inline README.md:316]: Can we not have these new line edits (this and many others in this PR) if there is not edit to the context. Thanks!

**Comment by jackgerrits** [comment]
If you'd like to revisit this please feel free to reopen with the requested changes made.

**Pull request closed without merging**